### ISSUE TITLE
Nomad fails to run job if using network mode bridge and dns options

### ISSUE THREAD

**Issue by danlsgiga** [hashicorp/nomad#8431]
### Nomad version
Nomad v0.12.0 (8f7fbc8e7b5a4ed0d0209968faf41b238e6d5817)

### Operating system and Environment details
`CentOS Linux release 7.7.1908 (Core)`

```
Client: Docker Engine - Community
 Version:           19.03.7
 API version:       1.40
 Go version:        go1.12.17
 Git commit:        7141c199a2
 Built:             Wed Mar  4 01:24:10 2020
 OS/Arch:           linux/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          19.03.12
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.13.10
  Git commit:       48a66213fe
  Built:            Mon Jun 22 15:45:28 2020
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.2.13
  GitCommit:        7ad184331fa3e55e52b890ea95e65ba581ae3429
 runc:
  Version:          1.0.0-rc10
  GitCommit:        dc9208a3303feef5b3839f4323d9beb36df0a9dd
 docker-init:
  Version:          0.18.0
  GitCommit:        fec3683
```
### Issue
Using the new `dns` block inside the `network` stanza with `mode = "bridge"` fails with:

```
failed to create container: API error (400): conflicting options: dns and the network mode
```

**Comment by spuder** [comment]
This is likely a duplicate of this issue. https://github.com/hashicorp/nomad/issues/7746

**Comment by danlsgiga** [comment]
Not sure, but was this fixed on 0.12.1?

**Comment by cthayer** [comment]
@danlsgiga It doesn't seem like it's fixed in 0.12.1 as I'm currently experiencing this error in my 0.12.1 cluster.
